debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/DateParser.java

```diff
@@ -0,0 +1,72 @@
+package io.debezium.connector.postgresql.connection.wal2json;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.format.TextStyle;
+import java.time.temporal.ChronoField;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DateParser {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DateParser.class);
+
+    private static final List<DateTimeFormatter> dateTimeFormatters = initializeDateTimeFormatters();
+
+    private static List<DateTimeFormatter> initializeDateTimeFormatters() {
+        // Postgresql years can be as large as 6 digits. See https://www.postgresql.org/docs/9.6/datatype-datetime.html
+        // Try parsing it with 4 digits first since that will be the common case. Then try up to 6 digits.
+        List<Integer> yearDigitsOrder = Arrays.asList(4, 1, 2, 3, 5, 6);
+
+        return yearDigitsOrder
+                .stream()
+                .map(DateParser::createDateTimeFormatter)
+                .collect(Collectors.toList());
+    }
+
+    private static DateTimeFormatter createDateTimeFormatter(int yearDigits) {
+        StringBuilder pattern = new StringBuilder();
```

Comment:
@jaydeesimon 
Would it be possible to constructer formatter using `java.time.format.DateTimeFormatterBuilder.appendValue(TemporalField, int, int, SignStyle)`? You'd pass 4 and 6 for the width of year?